primitives/arithmetic/src/lib.rs

```diff
@@ -85,8 +86,311 @@ where
 	}
 }
 
+/// A collection-like object that is made of values of type `T` and can normalize its individual
+/// values around a centric point.
+pub trait Normalizable<T> {
+	/// Normalize self around `t_max`.
+	///
+	/// Only returns `Ok` if the new sum of results is guaranteed to be equal to `t_max`. Else,
+	/// returns an error explaining why it failed to do so.
+	fn normalize(&self, t_max: T) -> Result<Vec<T>, &'static str>;
+}
+
+impl<T: Clone + Copy + Ord + BaseArithmetic + Debug> Normalizable<T> for Vec<T> {
```

Comment:
So here's a pickle: I want to implement this as such for all numeric types, and ideally have a different implementation for `impl<P: PerThing> Normalizable<P> for Vec<P>`, but understandably, rustc will fail here and cannot distinguish the fact that the trait bounds of the two blanket impls have nothing in common. 

This is understandable, you cannot assume two types are different just because their bounds are different. 

So for now, I implement this for `Vec<$name>` in the per_thing generation macro. The problem with this is that then it is actually not usable because in many places in the code we actually have `<P: PerThing> Vec<P>`. 

If anyone has a workaround for this, I'd appreciate it.

Reply:
One way I think is to change the trait as such: 

```rust
pub trait NormalizeAround<T> {
	/// Normalize the given collection around self.
	fn normalize(&self, collection: &[T]) -> Result<Vec<T>, &'static str> {
		normalize(collection, self)
	}
}
```

Reply:
As it turns out we actually now use only the impls for generic PerThing and concrete type u128, so I can add a blanket impl for PerThings and concrete impl for numeric types, and call it a day for now.